src/app/systelab-translate/date-util/date-util.ts

```diff
@@ -1,337 +1,348 @@
-import { addDays, format, setHours, setMilliseconds, setMinutes, setSeconds } from 'date-fns';
+import {addDays, format, setHours, setMilliseconds, setMinutes, setSeconds} from 'date-fns';
 
 export class DateUtil {
 
-	constructor(protected locale: string) {
-	}
+    constructor(protected locale: string) {
+    }
 
-	public setLocale(lang: string) {
-		this.locale = lang;
-	}
+    public setLocale(lang: string) {
+        this.locale = lang;
+    }
 
-	public getTimeFormat(withSeconds = false): string {
-		return (withSeconds) ? 'HH:mm:ss' : 'HH:mm';
-	}
+    public getTimeFormat(withSeconds = false, withAMPM = false): string {
+        const time = (withSeconds) ? 'HH:mm:ss' : 'HH:mm';
+        const marker = (withAMPM) ? ' a' : '';
+        return time + marker;
+    }
 
-	// DATE TIME FUNCTIONS
+    // DATE TIME FUNCTIONS
 
-	public formatDate(date: Date): string {
-		return format(date, this.getDateFormat());
+    public formatDate(date: Date): string {
+        return format(date, this.getDateFormat());
 
-	}
+    }
 
-	public formatDateFullYear(date: Date): string {
-		return format(date, this.getDateFormat(true));
+    public formatDateFullYear(date: Date): string {
+        return format(date, this.getDateFormat(true));
 
-	}
+    }
 
-	public formatTime(date: Date, withSeconds?: boolean): string {
-		return format(date, this.getTimeFormat(withSeconds));
-	}
+    public formatTime(date: Date, withSeconds?: boolean, withAMPM?: boolean): string {
+        return format(date, this.getTimeFormat(withSeconds, withAMPM));
+    }
 
-	public formatDateTime(date: Date, fullYear?: boolean, withSeconds?: boolean): string {
-		const formatedDate = (fullYear) ? this.formatDateFullYear(date) : this.formatDate(date);
-		const formatedHour = (withSeconds) ? this.formatTime(date, true) : this.formatTime(date);
-		return formatedDate + ' ' + formatedHour;
-	}
+    public formatDateTime(date: Date, fullYear?: boolean, withSeconds?: boolean, withAMPM?: boolean): string {
+        const formatedDate = (fullYear) ? this.formatDateFullYear(date) : this.formatDate(date);
+        let formatedHour;
+       if (withSeconds && withAMPM) {
```

Comment:
This block if/else can be simplified to : this.formatTime(date, withSeconds, withAMPM);